src/PostgREST/App.hs

```diff
@@ -1,100 +1,124 @@
+
+import           PostgREST.Auth
+import           PostgREST.Config          (AppConfig (..))
+import           PostgREST.Parsers
+import           PostgREST.PgQuery
+import           PostgREST.PgStructure
+import           PostgREST.QueryBuilder
+import           PostgREST.RangeQuery
+import           PostgREST.Types
+
+import           Prelude
+
+app :: DbStructure -> AppConfig -> BL.ByteString -> DbRole -> Request -> H.Tx P.Postgres s Response
+app dbstructure conf reqBody dbrole req =
   case (path, verb) of
+
     ([], _) -> do
-      body <- encode <$> tables (cs schema)
+      let body = encode $ filter (filterTableAcl dbrole) $ filter ((cs schema==).tableSchema) allTabs
       return $ responseLBS status200 [jsonH] $ cs body
 
     ([table], "OPTIONS") -> do
-      let qt = qualify table
-      cols <- columns qt
-      pkey <- map cs <$> primaryKeyColumns qt
-      return $ responseLBS status200 [jsonH, allOrigins]
-        $ encode (TableOptions cols pkey)
+      let cols = filter (filterCol schema table) allCols
+          pkeys = map pkName $ filter (filterPk schema table) allPrKeys
+          body = encode (TableOptions cols pkeys)
+      return $ responseLBS status200 [jsonH, allOrigins] $ cs body
 
     ([table], "GET") ->
       if range == Just emptyRange
       then return $ responseLBS status416 [] "HTTP Range error"
-      else do
-        let qt = qualify table
-            from = fromMaybe 0 $ rangeOffset <$> range
-            count = if hasPrefer "count=none" 
+      else
+        case queries of
+          Left e -> return $ responseLBS status400 [("Content-Type", "text/plain")] $ cs e
+          Right (qs, cqs) -> do
+            let qt = qualify table
+                count = if hasPrefer "count=none"
                       then countNone
-                      else countRows qt
-            query = B.Stmt "select " V.empty True <>
-                parentheticT (
-                  whereT qt qq count
-                ) <> commaq <> (
-                bodyForAccept contentType qt
-                . limitT range
-                . orderT (orderParse qq)
-                . whereT qt qq
-                $ select qt qq
-              )
-        row <- H.maybeEx query
-        let (tableTotal, queryTotal, body) =
-              fromMaybe (Just 0, 0, Just "" :: Maybe Text) row
-            to = from+queryTotal-1
-            contentRange = contentRangeH from to tableTotal
-            status = rangeStatus from to tableTotal
-            canonical = urlEncodeVars
-              . sortBy (comparing fst)
-              . map (join (***) cs)
-              . parseSimpleQuery
-              $ rawQueryString req
-        return $ responseLBS status
-          [contentTypeH, contentRange,
-            ("Content-Location",
-              "/" <> cs table <>
-                if Prelude.null canonical then "" else "?" <> cs canonical
-            )
-          ] (cs $ fromMaybe "[]" body)
+                      else cqs
+
```

Comment:
It is here : 

https://github.com/ruslantalpa/postgrest/blob/master/src/PostgREST/App.hs#L86

But well it is just a space so don't worry ;-)